index.html

```diff
@@ -34,6 +34,19 @@ <h3>Connect</h3>
     </ul>
   </section>
   <img src="./myself.jpg" alt="Picture of Me" width="100" height="90" />
+  <section>
+    <h2>Leave a Message</h2>
+    <label for="name">Name</label>
+    <form name="leave_message"><input type="text" name="name" required="true"><br><label for="mail">Email
+        Address</label><br><input type="email" name="email" required="true">
+      <br><label for="message">Messages</label><br><textarea name="message"
+        required="true"></textarea></label><br><label><button type="submit">Submit</button></label><br>
+    </form>
+  </section>
+  <section id="messages">
+    <h2>Messages</h2>
+    <ul></ul>
+  </section>
```

Comment:
Looking good. I know there were some issues with getting the formatted working in your VsCode, so remember you can use sites like https://codebeautify.org/htmlviewer to format your HTML to make it a bit easier to read

```html
<section>
      <h2>Leave a Message</h2>
      <label for="name">Name</label>
      <form name="leave_message">
        <input type="text" name="name" required="true">
        <br>
        <label for="mail">Email Address</label>
        <br>
        <input type="email" name="email" required="true">
        <br>
        <label for="message">Messages</label>
        <br>
        <textarea name="message" required="true"></textarea>
        </label>
        <br>
        <label>
          <button type="submit">Submit</button>
        </label>
        <br>
      </form>
    </section>
    <section id="messages">
      <h2>Messages</h2>
      <ul></ul>
    </section>
```

Reply:
There we can see a bit more clearly that `<label for="name">Name</label>` right now is _outside_ of your form elemen but it should be _**inside**_. In this particular case, it's not breaking anything but if you wanted to style things inside the form more consistently with CSS it could lead to some trouble down the road.

Reply:
The `for=` part of a `label` element should match up with the **id** attribute of an input or other type of form control element.

So for example for the email you have:
```html
        <label for="mail">Email Address</label>
        <br>
        <input type="email" name="email" required="true">
```

but to tell HTML properly that the label is associated with that input element, we need to give the input element a matching id attribute like this:

```html
        <label for="mail">Email Address</label>
        <br>
        <input type="email" name="email" required="true" id="mail">
```

The main benefit of this is that it'll make the labels clickable. So users can either click right on the input box or just on the label and start typing

Reply:
Labels should only wrap form elements that need to be labeled or that will be used for user input. So you don't need to wrap the button element in a label tag:

```html
        <label>
          <button type="submit">Submit</button>
        </label>
```

should just be

```html
<button type="submit">Submit</button>
```

Reply:
There seems to be an extra ` </label>` after the closing `</textarea>` tag, that's not actually needed because it doesn't have matching opening `<label>` tag